src/concurrency/thread.rs

```diff
@@ -730,19 +744,24 @@ impl<'tcx> ThreadManager<'tcx> {
             .threads
             .iter_enumerated()
             .skip(self.active_thread.index() + 1)
-            .chain(self.threads.iter_enumerated().take(self.active_thread.index()));
-        for (id, thread) in threads {
-            debug_assert_ne!(self.active_thread, id);
-            if thread.state.is_enabled() {
-                info!(
-                    "---------- Now executing on thread `{}` (previous: `{}`) ----------------------------------------",
-                    self.get_thread_display_name(id),
-                    self.get_thread_display_name(self.active_thread)
-                );
-                self.active_thread = id;
-                break;
-            }
+            .chain(self.threads.iter_enumerated().take(self.active_thread.index() + 1))
+            .filter(|(_id, thread)| thread.state.is_enabled())
+            .collect::<Vec<_>>();
+
+        let mut new_thread = threads.first();
```

Comment:
Also here, please don't use an initialize-then-mutate pattern. It's almost always better to initialize things with the right value immediately and eschew mutation. So here that would be `let new_thread = if self.round_robin_scheduling { ... } else { ... }`.